### PR TITLE
Drop GEOS 3.7, add GEOS 3.12

### PR DESCRIPTION
**Pull request by caspervdw**
Follow-up from https://github.com/shapely/shapely/pull/1453

Ubuntu 18 is now EOL, so 3.7 can be dropped.

### CODE REVIEW

**Comment by caspervdw** [comment]
I have been reading up on https://github.com/shapely/shapely/issues/1670 and https://github.com/shapely/shapely/pull/1708 considering the backwards incompatible change in `oriented_envelope` in GEOS 3.12

My personal opinion is that shapely shouldn't attempt to cover up backwards incompatible changes in GEOS. We should do nothing short of documenting that it happened. 

In this specific case we already had a solution in Shapely 1.x, so porting that back for GEOS 3.11 users could be done, however I think it is a very small user base that doesn't want to go forward to GEOS 3.12 so I think it is better to go forward and not fix anything at all.

File: shapely/constructive.py [hunk 1]
```diff
@@ -971,16 +969,16 @@ def voronoi_polygons(
     --------
     >>> from shapely import LineString, MultiPoint, normalize, Point
```
Comment (mwtoews) [inline shapely/constructive.py:970]:
```suggestion
    >>> from shapely import LineString, MultiPoint, Point
```

**Review by mwtoews — APPROVED** [review]
Thanks for putting this together!

There is still one hidden failure with GEOS main from your upstream fixes:

    pytest shapely/tests/test_creation.py::test_points_nan_becomes_empty

Is it worth undoing f115e06adad5e5d494d6d9b847a65c47d042734c ?

**Comment by caspervdw** [comment]
Thanks for the review @mwtoews . I’ll finish this up and then merge.

about the NaN points: I was planning to have a good look at the current situation when finishing https://github.com/shapely/shapely/pull/1811

**Pull request merged**